spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -290,6 +311,9 @@ protected DatagramSocket getTheSocket() {
 	}
 
 	protected synchronized DatagramSocket getSocket() throws IOException {
+		if (this.socketExpression != null) {
+			return this.socketExpression.getValue(this.evaluationContext, DatagramSocket.class);
```

Comment:
Yes - I did a notice about it in my last comment (in GitHubs Conversation tab).
"I watched AmqpOutboundEndpoint and other classes using expressions but I don't quite fill how to use ExpressionEvaluatingMessageProcessor with message oriented processing approach here?"